frontend/apps/reader/modules/readerstatus.lua

```diff
@@ -198,11 +198,9 @@ function ReaderStatus:deleteFile(file, text_end_book)
         ok_text = _("Delete"),
         ok_callback = function()
             local FileManager = require("apps/filemanager/filemanager")
-            local filemanagerutil = require("apps/filemanager/filemanagerutil")
             self.ui:onClose()
             FileManager:deleteFile(file)
-            filemanagerutil.removeFileFromHistoryIfWanted(file)
-            filemanagerutil.ensureLastFileExists()
+            require("readhistory"):fileDeleted(file) -- (will update "lastfile")
```

Comment:
is it `isFileDeleted()` check ? or `deleteFile()` ? It's not quite sure what it does before reading full implementation.

How about replacing the comments (as "Uncle Bob" recommends in his book `Clean code`) with self explanatory method name? so here for you name whould be something like:
`fileDeletedAndUpdateLastFile()` ?

So we get rid of comments that after some time are usually orphaned and only mislead devs..

Reply:
That it also happens to update `lastfile` is an irrelevant internal detail.

Reply:
> is it isFileDeleted() check ? or deleteFile() ? It's not quite sure what it does before reading full implementation.

None of that. It's just a notification that `thisFileHasJustBeenDeleted(file)`. Not a question, not a request for the action to delete it.
I thought about calling it `onFileDeleted()` - but it might invite people to think this might work as an UI event - and I'm not sure it does/will in any/all conditions.

> How about replacing the comments (as "Uncle Bob" recommends in his book Clean code) with self explanatory method name? so here for you name whould be something like:
fileDeletedAndUpdateLastFile() ?

Well, the idea is that you shouldn't care much about what ReadHistory might do (now and in the future). So you just tell it that this file has been fileDeleted() - and let it do what it should do.
(and yes, the curious dev will have to look at the implementation to know what's done or not - but I expect any serious dev to be that curious :)

Reply:
> I thought about calling it onFileDeleted() - but it might invite people to think this might work as an UI event - and I'm not sure it does/will in any/all conditions.

You could consider something similar in meaning like process/notify that doesn't have the event handling baggage.

Reply:
Any suggestion? (I thought the `'ed` suffix was enough to convey that meaning).

Reply:
Well, I'd have said something if I didn't think it was fine. But `notifyFileDeleted` would make it slightly more explicit.

Reply:
Actually scratch that, I like that much less. Please just keep it. ;-)